sway/input/cursor.c

```diff
@@ -485,6 +560,30 @@ void dispatch_cursor_button(struct sway_cursor *cursor,
 	double sx, sy;
 	struct sway_container *cont = container_at_coords(cursor->seat,
 			cursor->cursor->x, cursor->cursor->y, &surface, &sx, &sy);
+
+	// Handle mouse bindings
+	bool on_border = find_resize_edge(cont, cursor) != WLR_EDGE_NONE;
+	bool on_contents = !on_border && surface;
+	bool on_titlebar = !on_border && !surface;
+	struct sway_binding *binding = NULL;
+	struct wlr_keyboard *keyboard = wlr_seat_get_keyboard(cursor->seat->wlr_seat);
+	uint32_t modifiers = wlr_keyboard_get_modifiers(keyboard);
```

Comment:
What happens if no keyboard is connected? Wouldn't `keyboard` be `NULL` and cause a crash?

Reply:
Indeed. Fixed.